RecoTracker/LSTCore/src/alpaka/TrackCandidate.h

```diff
@@ -63,50 +86,82 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
-                                                                  uint8_t* logicalLayerIndices,
-                                                                  uint16_t* lowerModuleIndices,
-                                                                  unsigned int* hitIndices,
-                                                                  int pixelSeedIndex,
-                                                                  float centerX,
-                                                                  float centerY,
-                                                                  float radius,
-                                                                  unsigned int trackCandidateIndex,
-                                                                  unsigned int directObjectIndex) {
-    candsBase.trackCandidateType()[trackCandidateIndex] = trackCandidateType;
-    candsExtended.directObjectIndices()[trackCandidateIndex] = directObjectIndex;
-    candsBase.pixelSeedIndex()[trackCandidateIndex] = pixelSeedIndex;
-
-    candsExtended.objectIndices()[trackCandidateIndex][0] = innerTrackletIndex;
-    candsExtended.objectIndices()[trackCandidateIndex][1] = outerTrackletIndex;
+    // T5: 5 OT logical layers in [1..11], mapped to layer slots (logicalLayer - 1) + 2
+    if (trackCandidateType == LSTObjType::T5) {
+      CMS_UNROLL_LOOP for (int layerIndex = 0; layerIndex < Params_T5::kLayers; ++layerIndex) {
+        const uint8_t logicalLayer = logicalLayerIndices[layerIndex];  // 1..11
+        const int layerSlot = (logicalLayer - 1) + 2;                  // OT slots
+        addTrackCandidateLayerHits(candsBase,
+                                   candsExtended,
+                                   trackCandidateIndex,
+                                   layerSlot,
+                                   logicalLayer,
+                                   lowerModuleIndices[layerIndex],
+                                   hitIndices[2 * layerIndex + 0],
+                                   hitIndices[2 * layerIndex + 1]);
+      }
+    } else if (trackCandidateType == LSTObjType::pT5) {
+      // Pixel layers occupy slots 0 and 1 with logicalLayer = 0
+      addTrackCandidateLayerHits(
+          candsBase, candsExtended, trackCandidateIndex, 0, 0, lowerModuleIndices[0], hitIndices[0], hitIndices[1]);
+      addTrackCandidateLayerHits(
+          candsBase, candsExtended, trackCandidateIndex, 1, 0, lowerModuleIndices[1], hitIndices[2], hitIndices[3]);
+
+      // pT5 then carries the T5's 5 OT layers starting at logicalLayerIndices[2]; hits start at index 4
+      CMS_UNROLL_LOOP for (int layerIndex = 2; layerIndex < Params_pT5::kLayers; ++layerIndex) {
+        const uint8_t logicalLayer = logicalLayerIndices[layerIndex];  // 1..11
+        const int layerSlot = (logicalLayer - 1) + 2;                  // OT slots
+        const int hitOffset = 4 + 2 * (layerIndex - 2);                // 10 OT hits after 4 pixel hits
```

Comment:
need to define smth like `kFirstOTLayer` in the `Params_pT5` and such;
then replace corresponding `2` by `kHitsPerLayer` and `kFirstOTLayer` to avoid confusion